### PR TITLE
WIP Add the theta neural mass models

### PR DESCRIPTION
**Pull request by i-Zaak**
Here we add implementations of various neural mass models derived by Ott-Antonsen reduction of infinitely all-to-all coupled QIF neurons (Theta-neurons).

### CODE REVIEW

**Comment by i-Zaak** [comment]
Todo:

- [ ] add demo notebook(s)

File: scientific_library/tvb/simulator/models/infinite_theta.py [hunk 1]
```diff
@@ -0,0 +1,871 @@
...
+    2D model describing the Ott-Antonsen reduction of infinitely all-to-all
+    coupled QIF neurons (Theta-neurons) as in Montbrio, Pazo, Roxin, 2015.
+
+    The two state variables :math:`r` and :math:`V` represent the average
+    firing rate and the average membrane potential of our QIF neurons.
+
+    The equations of the infinite QIF 2D population model read
+
+    .. math::
+            \dot{r} &= 1/tau (Delta/(pi tau) + 2 V r),
```
Comment (maedoc) [inline scientific_library/tvb/simulator/models/infinite_theta.py:18]: why is there a `,` at the end of the line? (have this question in other model docstrings too)

File: scientific_library/tvb/simulator/models/infinite_theta.py [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+    )
+
+
+    state_variables = ('r', 'V', 'g', 'q')
+    _nvar = 4
+    # Cvar is the coupling variable. 
+    cvar = numpy.array([0,1,2,3], dtype=numpy.int32)
+    
+    def dfun(self, state_variables, coupling, local_coupling=0.0):
+        r = state_variables[0, :]
```
Comment (maedoc) [inline scientific_library/tvb/simulator/models/infinite_theta.py:234]: `r, V, g, q = state_variables` is a little nicer, also ensures `state_variables.shape[0]==self._nvar`

File: scientific_library/tvb/simulator/models/infinite_theta.py (hunk 1 above)
Comment (i-Zaak) [reply scientific_library/tvb/simulator/models/infinite_theta.py:18]: This is an artifact of copy-pasting the docstring from other model. Do we have a quick way to render these?

Comment (i-Zaak) [reply scientific_library/tvb/simulator/models/infinite_theta.py]: Ok, Spyder seems to be a decent tool to edit the docstrings.

**Comment by i-Zaak** [comment]
All fixed and polished.

**Review by maedoc — APPROVED** [review]
Looks great, thx

**Comment by popaula937** [comment]
I suppose these models should also be displayed in the GUI. We could add a task for that in the next sprint, or would this be more urgent?

**Comment by i-Zaak** [comment]
Yes, GUI is not urgent right now, next sprint is ok.

**Pull request merged**